data-prepper-plugins/sqs-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/sqs/SqsSink.java

```diff
@@ -0,0 +1,122 @@
+import org.opensearch.dataprepper.aws.api.AwsConfig;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsOptions;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.sink.AbstractSink;
+import org.opensearch.dataprepper.model.sink.Sink;
+import org.opensearch.dataprepper.model.sink.SinkContext;
+import org.opensearch.dataprepper.plugins.source.sqs.common.SqsClientFactory;
+import org.opensearch.dataprepper.model.codec.OutputCodec;
+import org.opensearch.dataprepper.plugins.dlq.DlqPushHandler;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.regions.Region;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Collection;
+
+@Experimental
+@DataPrepperPlugin(name = "sqs", pluginType = Sink.class, pluginConfigurationType = SqsSinkConfig.class)
+public class SqsSink extends AbstractSink<Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SqsSink.class);
+    private static final Duration RETRY_FLUSH_BACKOFF = Duration.ofSeconds(5);
+    private final SqsSinkConfig sqsSinkConfig;
+    private volatile boolean sinkInitialized;
+    private final SqsSinkService sqsSinkService;
+
+    @DataPrepperPluginConstructor
+    public SqsSink(final PluginSetting pluginSetting,
+                   final PluginMetrics pluginMetrics,
+                   final PluginFactory pluginFactory,
+                   final SqsSinkConfig sqsSinkConfig,
+                   final SinkContext sinkContext,
+                   final ExpressionEvaluator expressionEvaluator,
+                   final AwsCredentialsSupplier awsCredentialsSupplier) {
+        super(pluginSetting);
+        this.sqsSinkConfig = sqsSinkConfig;
+        sinkInitialized = false;
+        final PluginModel codecConfiguration = sqsSinkConfig.getCodec();
+        final PluginSetting codecPluginSettings;
+        if (codecConfiguration != null) {
+            String codecPluginName = codecConfiguration.getPluginName();
+            if (!codecPluginName.equals("json") && !codecPluginName.equals("ndjson")) {
+                throw new RuntimeException(String.format("Codec {} not supported.", codecPluginName));
+            }
+            codecPluginSettings = new PluginSetting(codecConfiguration.getPluginName(),
+                codecConfiguration.getPluginSettings());
+        } else {
+            codecPluginSettings = new PluginSetting("ndjson", Map.of());
```

Comment:
We should also require that the threshold `max_events_per_message` is set in this case.

This could should then set the `max_events_per_message` to `1` because that is what this would be.